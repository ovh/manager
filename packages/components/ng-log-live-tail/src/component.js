import template from './template.html';
import controller from './controller';

export default {
  controller,
  template,
  transclude: {
    tile: '?tile',
    description: '?description',
  },
  bindings: {
    logApiUrl: '<',
    logKind: '<',
    logKeys: '<',
  },
};
