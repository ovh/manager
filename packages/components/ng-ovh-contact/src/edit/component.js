import controller from './controller';
import template from './template.html';

export default {
  require: {
    ovhContactCtrl: '^ovhContact',
  },
  template,
  controller,
  bindings: {
    options: '=?ovhContactChoiceOptions',
    customList: '=?ovhContactChoiceCustomList',
  },
};
