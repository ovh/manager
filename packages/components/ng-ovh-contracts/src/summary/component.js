import template from './template.html';

const component = {
  bindings: {
    contracts: '<items',
    inlineContracts: '<inlineItems',
    model: '=',
    name: '@?',
  },
  template,
};

export default component;
