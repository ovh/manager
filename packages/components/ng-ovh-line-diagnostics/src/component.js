import controller from './controller';
import template from './template.html';

export default {
  bindings: {
    lineNumber: '@',
    serviceName: '@',
  },
  template,
  controller,
};
