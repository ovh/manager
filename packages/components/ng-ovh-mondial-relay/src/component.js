import controller from './controller';
import template from './template.html';

export default {
  bindings: {
    ngModel: '=?',
    userService: '=?',
    mondialRelayService: '=?',
  },
  template,
  controller,
};
