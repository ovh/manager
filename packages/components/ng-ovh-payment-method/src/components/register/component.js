import controller from './controller';
import template from './index.html';

export default {
  name: 'ovhPaymentMethodRegister',
  controller,
  template,
  bindings: {
    defaultPaymentMethodType: '@?',
    defaultPaymentMethodIntegration: '@?',
    model: '<',
    modelName: '@?',
    automaticSelect: '<?',
    automaticDefault: '<?',
    paymentMethodTypesOrder: '<?',
    paymentMethodTypesPerLine: '<?',
    registeredPaymentMethods: '<?',
    unregisteredPaymentMethods: '<?',
    excludedPaymentMethods: '<?',
    showSetAsDefaultChoice: '<?',
    forceSetAsDefaultChoice: '<?',
    onInitializationError: '&?',
    onInitialized: '&?',
    showDefaultExplanationTexts: '<?',
    defaultExplanationTextsOptions: '<?',
    required: '<?',
  },
  transclude: {
    introductionText: '?ovhPaymentMethodRegisterIntroductionText',
    explanationText: '?ovhPaymentMethodRegisterExplanationText',
    creditCardExplanationText:
      '?ovhPaymentMethodRegisterCreditCardExplanationText',
    sepaExplanationText: '?ovhPaymentMethodRegisterSepaExplanationText',
    paypalExplanationText: '?ovhPaymentMethodRegisterPaypalExplanationText',
    rupayExplanationText: '?ovhPaymentMethodRegisterRupayExplanationText',
  },
};
