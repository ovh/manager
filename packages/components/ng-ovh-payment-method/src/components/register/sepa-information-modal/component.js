import template from './template.html';

export default {
  template,
  bindings: {
    onConfirm: '&',
    isModalOpened: '<',
  },
};
