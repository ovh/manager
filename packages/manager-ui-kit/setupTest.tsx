import * as React from 'react';
import { ComponentType } from 'react';
import { I18nextProvider, initReactI18next } from 'react-i18next';
import i18n from 'i18next';
import {
  RenderOptions,
  RenderResult,
  render,
  cleanup,
} from '@testing-library/react';
import '@testing-library/jest-dom';
import 'element-internals-polyfill';
import { vi } from 'vitest';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import ActionMenuTransFR from './src/components/action-menu/translations/Messages_fr_FR.json';
import DatagridTransFR from './src/components/datagrid/translations/Messages_fr_FR.json';
import FiltersTransFR from './src/components/filters/translations/Messages_fr_FR.json';

i18n.use(initReactI18next).init({
  fallbackLng: 'fr_FR',
  interpolation: {
    escapeValue: false,
  },
  resources: {
    fr: {
      'action-menu': {
        ...ActionMenuTransFR,
      },
      datagrid: {
        ...DatagridTransFR,
      },
      filterAdd: {
        ...FiltersTransFR,
      },
    },
  },
});

export const queryClient = new QueryClient({
  defaultOptions: {
    mutations: {
      retry: false,
    },
    queries: {
      retry: false,
    },
  },
});

const Wrappers = ({ children }: { children: React.ReactElement }) => {
  return (
    <QueryClientProvider client={queryClient}>
      <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
    </QueryClientProvider>
  );
};

const customRender = (
  ui: React.JSX.Element,
  options?: Omit<RenderOptions, 'queries'>,
): RenderResult =>
  render(ui, { wrapper: Wrappers as ComponentType, ...options });

export { customRender as render, cleanup };

// This polyfill exists because of an issue with jsdom and the EventTarget class
// when testing a component with an OdsDatepicker (addEventListener crashes at component initialization).
// Fix from issue: https://github.com/jsdom/jsdom/issues/2156
global.EventTarget = class {
  listeners = {};

  addEventListener(type, listener) {
    this.listeners = this.listeners || {};
    (this.listeners[type] || (this.listeners[type] = new Set())).add(listener);
  }

  removeEventListener(type, listener) {
    if (this.listeners && this.listeners[type]) {
      this.listeners[type].delete(listener);
    }
  }

  dispatchEvent(event) {
    this.listeners[event.type].forEach((listener) => listener(event));
    return !event.defaultPrevented;
  }
};

const ResizeObserverMock = vi.fn((callback) => {
  // Create a mock ResizeObserverEntry with the expected structure
  const mockEntry = {
    target: document.createElement('div'),
    contentRect: {
      width: 100,
      height: 100,
      top: 0,
      left: 0,
      bottom: 100,
      right: 100,
      x: 0,
      y: 0,
    },
    borderBoxSize: [{ width: 100, height: 100 }],
    contentBoxSize: [{ width: 100, height: 100 }],
    devicePixelContentBoxSize: [{ width: 100, height: 100 }],
  };

  // Call the callback with an array of entries as the real ResizeObserver does
  callback([mockEntry]);

  return {
    observe: vi.fn(),
    unobserve: vi.fn(),
    disconnect: vi.fn(),
  };
});

vi.stubGlobal('ResizeObserver', ResizeObserverMock);

// Mock getBoundingClientRect for virtualization and components
Element.prototype.getBoundingClientRect = vi.fn(() => ({
  width: 1000,
  height: 600,
  top: 0,
  left: 0,
  bottom: 600,
  right: 1000,
  x: 0,
  y: 0,
  toJSON: () => {},
}));

// Mock HTMLElement methods for components
HTMLElement.prototype.scrollTo = vi.fn();
HTMLElement.prototype.scrollIntoView = vi.fn();

// Mock scroll properties needed by ODS Modal
Object.defineProperty(Element.prototype, 'scrollLeft', {
  configurable: true,
  writable: true,
  value: 0,
});

Object.defineProperty(Element.prototype, 'scrollTop', {
  configurable: true,
  writable: true,
  value: 0,
});

Object.defineProperty(Element.prototype, 'scrollWidth', {
  configurable: true,
  value: 0,
});

Object.defineProperty(Element.prototype, 'scrollHeight', {
  configurable: true,
  value: 0,
});

// Suppress unhandled async errors from ODS Modal cleanup
const originalConsoleError = console.error;
console.error = (...args: any[]) => {
  const errorMessage = args[0]?.toString() || '';
  // Suppress specific ODS Modal/Zag.js cleanup errors
  if (
    errorMessage.includes("reading 'left'") ||
    errorMessage.includes('[@zag-js/dismissable]') ||
    errorMessage.includes('preventScroll') ||
    errorMessage.includes('Uncaught Exception')
  ) {
    return;
  }
  originalConsoleError(...args);
};

// Catch unhandled errors from ODS Modal async cleanup
if (typeof window !== 'undefined') {
  window.addEventListener('error', (event) => {
    const errorMessage = event.message || event.error?.message || '';
    // Suppress ODS Modal cleanup errors
    if (
      errorMessage.includes("reading 'left'") ||
      errorMessage.includes('[@zag-js/dismissable]')
    ) {
      event.preventDefault();
      event.stopPropagation();
      return false;
    }
    return true;
  });

  window.addEventListener('unhandledrejection', (event) => {
    const errorMessage = event.reason?.message || '';
    // Suppress ODS Modal cleanup errors
    if (
      errorMessage.includes("reading 'left'") ||
      errorMessage.includes('[@zag-js/dismissable]')
    ) {
      event.preventDefault();
      event.stopPropagation();
    }
  });
}
