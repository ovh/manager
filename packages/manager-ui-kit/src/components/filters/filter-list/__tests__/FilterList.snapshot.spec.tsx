import { vi } from 'vitest';

import { FilterList } from '@/components/filters/filter-list/FilterList.component';
import { render } from '@/setupTest';

import { FilterListProps } from '../FilterList.props';

const renderComponent = (props: FilterListProps) => {
  return render(<FilterList {...props} />);
};

describe('FilterList Snapshot Tests', () => {
  it('should match snapshot with empty filters', () => {
    const props = {
      filters: [],
      onRemoveFilter: vi.fn(),
    } as FilterListProps;

    const { container } = renderComponent(props);
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with single string filter', () => {
    const props = {
      filters: [
        {
          key: 'username',
          comparator: 'includes',
          value: 'john_doe',
          label: "Nom d'utilisateur",
          type: 'String',
        },
      ],
      onRemoveFilter: vi.fn(),
    } as FilterListProps;

    const { container } = renderComponent(props);
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with date filter', () => {
    const props = {
      filters: [
        {
          key: 'createdAt',
          comparator: 'is_equal',
          value: new Date('2023-10-15').toISOString(),
          label: 'Date de crÃ©ation',
          type: 'Date',
        },
      ],
      onRemoveFilter: vi.fn(),
    } as FilterListProps;

    const { container } = renderComponent(props);
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with numeric filter', () => {
    const props = {
      filters: [
        {
          key: 'age',
          comparator: 'is_higher',
          value: '25',
          label: 'Age',
          type: 'Numeric',
        },
      ],
      onRemoveFilter: vi.fn(),
    } as FilterListProps;

    const { container } = renderComponent(props);
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with multiple filters', () => {
    const props = {
      filters: [
        {
          key: 'username',
          comparator: 'includes',
          value: 'admin',
          label: 'Username',
          type: 'String',
        },
        {
          key: 'status',
          comparator: 'is_equal',
          value: 'active',
          label: 'Status',
        },
        {
          key: 'createdAt',
          comparator: 'is_after',
          value: new Date('2023-01-01').toISOString(),
          label: 'Created After',
          type: 'Date',
        },
      ],
      onRemoveFilter: vi.fn(),
    } as FilterListProps;

    const { container } = renderComponent(props);
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with different comparators', () => {
    const props = {
      filters: [
        {
          key: 'name',
          comparator: 'starts_with',
          value: 'Test',
          label: 'Name',
        },
        {
          key: 'email',
          comparator: 'ends_with',
          value: '@example.com',
          label: 'Email',
        },
        {
          key: 'role',
          comparator: 'is_different',
          value: 'guest',
          label: 'Role',
        },
      ],
      onRemoveFilter: vi.fn(),
    } as FilterListProps;

    const { container } = renderComponent(props);
    expect(container).toMatchSnapshot();
  });
});
