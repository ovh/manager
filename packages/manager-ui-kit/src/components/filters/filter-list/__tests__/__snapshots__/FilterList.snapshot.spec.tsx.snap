// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FilterList Snapshot Tests > should match snapshot with date filter 1`] = `
<div>
  <button
    class="_tag_t7a4o_6 _tag--information_t7a4o_35 _tag--lg_t7a4o_114 mr-3"
    data-ods="tag"
    data-testid="filter-list_tag_item"
    type="button"
  >
    Date de création est égal à 15/10/2023
    <span
      class="_icon_10c23_2 _icon--xmark_10c23_562 _tag__close_t7a4o_111"
      data-ods="icon"
      role="presentation"
    />
  </button>
</div>
`;

exports[`FilterList Snapshot Tests > should match snapshot with different comparators 1`] = `
<div>
  <button
    class="_tag_t7a4o_6 _tag--information_t7a4o_35 _tag--lg_t7a4o_114 mr-3"
    data-ods="tag"
    data-testid="filter-list_tag_item"
    type="button"
  >
    Name débute par Test
    <span
      class="_icon_10c23_2 _icon--xmark_10c23_562 _tag__close_t7a4o_111"
      data-ods="icon"
      role="presentation"
    />
  </button>
  <button
    class="_tag_t7a4o_6 _tag--information_t7a4o_35 _tag--lg_t7a4o_114 mr-3"
    data-ods="tag"
    data-testid="filter-list_tag_item"
    type="button"
  >
    Email termine par @example.com
    <span
      class="_icon_10c23_2 _icon--xmark_10c23_562 _tag__close_t7a4o_111"
      data-ods="icon"
      role="presentation"
    />
  </button>
  <button
    class="_tag_t7a4o_6 _tag--information_t7a4o_35 _tag--lg_t7a4o_114 mr-3"
    data-ods="tag"
    data-testid="filter-list_tag_item"
    type="button"
  >
    Role est différent de guest
    <span
      class="_icon_10c23_2 _icon--xmark_10c23_562 _tag__close_t7a4o_111"
      data-ods="icon"
      role="presentation"
    />
  </button>
</div>
`;

exports[`FilterList Snapshot Tests > should match snapshot with empty filters 1`] = `<div />`;

exports[`FilterList Snapshot Tests > should match snapshot with multiple filters 1`] = `
<div>
  <button
    class="_tag_t7a4o_6 _tag--information_t7a4o_35 _tag--lg_t7a4o_114 mr-3"
    data-ods="tag"
    data-testid="filter-list_tag_item"
    type="button"
  >
    Username contient admin
    <span
      class="_icon_10c23_2 _icon--xmark_10c23_562 _tag__close_t7a4o_111"
      data-ods="icon"
      role="presentation"
    />
  </button>
  <button
    class="_tag_t7a4o_6 _tag--information_t7a4o_35 _tag--lg_t7a4o_114 mr-3"
    data-ods="tag"
    data-testid="filter-list_tag_item"
    type="button"
  >
    Status est égal à active
    <span
      class="_icon_10c23_2 _icon--xmark_10c23_562 _tag__close_t7a4o_111"
      data-ods="icon"
      role="presentation"
    />
  </button>
  <button
    class="_tag_t7a4o_6 _tag--information_t7a4o_35 _tag--lg_t7a4o_114 mr-3"
    data-ods="tag"
    data-testid="filter-list_tag_item"
    type="button"
  >
    Created After est après 01/01/2023
    <span
      class="_icon_10c23_2 _icon--xmark_10c23_562 _tag__close_t7a4o_111"
      data-ods="icon"
      role="presentation"
    />
  </button>
</div>
`;

exports[`FilterList Snapshot Tests > should match snapshot with numeric filter 1`] = `
<div>
  <button
    class="_tag_t7a4o_6 _tag--information_t7a4o_35 _tag--lg_t7a4o_114 mr-3"
    data-ods="tag"
    data-testid="filter-list_tag_item"
    type="button"
  >
    Age est supérieur à 25
    <span
      class="_icon_10c23_2 _icon--xmark_10c23_562 _tag__close_t7a4o_111"
      data-ods="icon"
      role="presentation"
    />
  </button>
</div>
`;

exports[`FilterList Snapshot Tests > should match snapshot with single string filter 1`] = `
<div>
  <button
    class="_tag_t7a4o_6 _tag--information_t7a4o_35 _tag--lg_t7a4o_114 mr-3"
    data-ods="tag"
    data-testid="filter-list_tag_item"
    type="button"
  >
    Nom d'utilisateur contient john_doe
    <span
      class="_icon_10c23_2 _icon--xmark_10c23_562 _tag__close_t7a4o_111"
      data-ods="icon"
      role="presentation"
    />
  </button>
</div>
`;
